polkadot/runtime/common/src/paras_registrar/mod.rs

```diff
@@ -431,6 +458,35 @@ pub mod pallet {
 			runtime_parachains::set_current_head::<T>(para, new_head);
 			Ok(())
 		}
+
+		#[pallet::call_index(9)]
+		#[pallet::weight(<T as Config>::WeightInfo::register())]
+		pub fn register_rental(
+			origin: OriginFor<T>,
+			id: ParaId,
+			genesis_head: HeadData,
+			validation_code: ValidationCode,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			Self::do_register(who, None, id, genesis_head, validation_code, true, false)?;
+			Ok(())
+		}
+
+		/// Callable by anyone.
+		#[pallet::call_index(10)]
+		#[pallet::weight(<T as Config>::WeightInfo::register())]
+		pub fn pay_rent(origin: OriginFor<T>, id: ParaId) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let mut rent_info = RentedParas::<T>::get(id).ok_or(Error::<T>::NotParathread)?;
+			<T as Config>::Currency::reserve(&who, rent_info.rent_cost)?;
+
+			let now = shared::Pallet::<T>::session_index();
+			rent_info.last_rent_payment = now;
```

Comment:
What if the next rent gets paid, before the old expired? Shouldn't this be set to `now + remaining`?

Reply:
Agreed, this should be set to `now + remaining`. (NOTE: This is still an early draft :))